### ISSUE TITLE
ipl_pin_list_direct syntax error when using partial nodes with local scheduler

### ISSUE THREAD

**Issue by boegel** [hpcugent/vsc-mympirun#117]
A warning like this appears when using the `local` scheduler on a partial node (e.g. ppn=16 on HPC-UGent's swalot, which has 20 cores per workernode):

```
IPL WARN> ipl_pin_list_direct syntax error, 16 list member should be -1, single CPU number, or CPU number range
```

For example (also occurs with `intel/2017a`):

```
$ module load intel/2016a
$ mympirun --sched=local hostname
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
IPL WARN> ipl_pin_list_direct syntax error, 16 list member should be -1, single CPU number, or CPU number range

$ echo $PBS_NUM_PPN
16

$ hostname
node2662.swalot.os
```

Does no occur with `pbs` scheduler:

```
$ mympirun --sched=pbs hostname
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2662.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
node2668.swalot.os
```

**Comment by boegel** [comment]
@Caylo debug output from `mympirun` for a situation in which the warning occurs is available at https://gist.github.com/boegel/890002ca94e18747401a0a1fd4320571

it seems like the warning is only printed in some cases, i.e. it depends on which cores get allocated

log messages that look very relevant:

```
2017-04-19 11:58:55,953 DEBUG      mympirun.affinity MainThread  sched_getaffinity for pid 20700 returned cpuset 11,13,15,17
2017-04-19 11:58:55,953 DEBUG      mympirun.IntelHydraMPIPbsdsh_Local MainThread  found cpus from affinity: [11, 13, 15, 17]
...
2017-04-19 11:58:55,953 INFO       mympirun.IntelHydraMPIPbsdsh_Local MainThread  check_usable_cpus: non-standard cpus found: found cpus 20, usable cpus 4
2017-04-19 11:58:55,953 INFO       mympirun.IntelHydraMPIPbsdsh_Local MainThread  check_usable_cpus: one node requested. Setting I_MPI_PIN_PROCESSOR_LIST to 11,13,15,17
...
IPL WARN> ipl_pin_list_direct syntax error, 11 list member should be -1, single CPU number, or CPU number range
```

see also https://software.intel.com/en-us/node/528898

**Comment by boegel** [comment]
for reference, the warning did not occur in other setups, e.g. 13 cores on a 16-core system with:

```
2017-04-19 12:03:57,903 DEBUG      mympirun.affinity MainThread  sched_getaffinity for pid 34033 returned cpuset 0-12
2017-04-19 12:03:57,903 DEBUG      mympirun.IntelHydraMPIPbsdsh_Local MainThread  found cpus from affinity: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
2017-04-19 12:03:57,903 INFO       mympirun.IntelHydraMPIPbsdsh_Local MainThread  check_usable_cpus: non-standard cpus found: found cpus 16, usable cpus 13
2017-04-19 12:03:57,904 INFO       mympirun.IntelHydraMPIPbsdsh_Local MainThread  check_usable_cpus: one node requested. Setting I_MPI_PIN_PROCESSOR_LIST to 0,1,2,3,4,5,6,7,8,9,10,11,12
```

**Comment by Caylo** [comment]
I_MPI_PIN_PROCESSOR_LIST should use logical core ids of cpus *within the cpu set*, not on the node itself. Not setting it or setting it wrongly results in no pinning, which is not desirable.
possible solution: detect if we are in a cpuset, and if yes, use _all_ (?)

relevant: https://software.intel.com/en-us/node/726664

```vsc40023@node2402:~ $ mympirun --sched=local ./testpin.sh
pid 24803's current affinity list: 6-8,18-20,22,23
pid 24804's current affinity list: 6-8,18-20,22,23
pid 24806's current affinity list: 6-8,18-20,22,23
pid 24807's current affinity list: 6-8,18-20,22,23
pid 24805's current affinity list: 6-8,18-20,22,23
pid 24808's current affinity list: 6-8,18-20,22,23
pid 24802's current affinity list: 6-8,18-20,22,23
pid 24809's current affinity list: 6-8,18-20,22,23
IPL WARN> ipl_pin_list_direct syntax error, 8 list member should be -1, single CPU number, or CPU number range```

**Comment by boegel** [comment]
Not setting `I_MPI_PIN_PROCESSOR_LIST` should be fine I think, then it'll just fall back to the default. So, if the default is decent (which I think it is), we should simply not set `I_MPI_PIN_PROCESSOR_LIST`.

Setting it incorrectly results in no (or wrong) pinning, which is definitely not what we want.

The above output was obtained with the following test script run under `mympirun`:

```shell
#!/bin/bash
taskset -c -p $$
```

Triggering the warning is a bit tricky because it only occurs when the cpuset you are in is *not* aligned with the first N available cores (so, it doesn't happen when your job is started on an empty node).

**Comment by boegel** [comment]
fixed in #122
